src/main/java/de/dhbw/karlsruhe/domain/services/dialogs/PlayDialogService.java

```diff
@@ -81,16 +88,44 @@ private void startGame() {
             }
         }
 
+        List<String> notCorrectFields = this.sudokuValidator.crossCheck(sudoku);
+        outputPort.notCorrectSudoku(notCorrectFields);
```

Comment:
Hier sollte vorher überprüft werden, ob alle Felder ausgefüllt sind. Sonst kommt die Nachricht, dass das Sudoku fertig ist auch, wenn man eig. nur das Sudoku geschlossen hat. Das liegt daran, dass break in der Schleife oben nicht die Methode beendet.
![image](https://github.com/mohjohfox/cli-sudoku/assets/92317814/3c1256cb-9167-475b-9a60-a4e16183d379)

Speichern für das Leaderboard weiter unten auch nur, wenn das Spielfeld wirklich voll ist.

Einfache Lösung wäre das Break; mit einem return; zu ersetzten oder die Abfrage sudokuValidator.isSudokuNotFullyFilled(sudoku.getGameField().sudokuArray()) für den Code-block wieder hinzuzufügen.

Reply:
Das ist immer noch der Fall... Weiter unten wird `if (!sudokuValidator.isSudokuNotFullyFilled(sudoku.getGameField().sudokuArray())) {
            List<String> notCorrectFields = this.sudokuValidator.crossCheck(sudoku);
            notCorrectFields = this.sudokuValidator.crossCheck(sudoku);
            outputPort.notCorrectSudoku(notCorrectFields);
        }`
aufgerufen, dadurch kommt es zu zwei mal der Ausgabe, wenn das Sudoku richtig gelöst wurde: 
![image](https://github.com/mohjohfox/cli-sudoku/assets/92317814/eac2e32e-ab55-4807-881e-c777de05cb88)
Und immer noch zu der Meldung welche Felder falsch sind, wenn das Sudoku nicht fertig ist...

Reply:
Bitte die zwei Zeilen hier oben einfach rauslöschen

Reply:
Oh, habe mich in den Zeilen vertan und gedacht es wäre schon behoben.